L32-node-channel-API.md

```diff
@@ -0,0 +1,103 @@
+
+### New `Channel` Class API
+
+```ts
+enum ConnectivityState {
+    IDLE = 0,
+    CONNECTING = 1,
+    READY = 2,
+    TRANSIENT_FAILURE = 3,
+    FATAL_FAILURE = 4
+}
+
+class Channel {
+    /**
+     * This constructor API is almost identical to the Client constructor,
+     * except that some of the options for the Client constructor are not valid
+     * here.
+     * @param target The address of the server to connect to
+     * @param credentials Channel credentials to use when connecting
+     * @param options A map of channel options that will be passed to the core
+     */
+    constructor(target: string, credentials: ChannelCredentials, options: ([key:string]: string|number));
+    /**
+     * Close the channel. This has the same functionality as the existing grpc.Client.prototype.close
+     */
+    close(): void;
+    /**
+     * Return the target that this channel connects to
+     */
+    getTarget(): string;
+    /**
+     * Get the channel's current connectivity state. This method is here mainly
+     * because it is in the existing internal Channel class, and there isn't
+     * another good place to put it.
+     * @param tryToConnect If true, the channel will start connecting if it is
+     *     idle. Otherwise, idle channels will only start connecting when a
+     *     call starts.
+     */
+    getConnectivityState(tryToConnect: boolean): ConnectivityState;
+    /**
+     * Watch for connectivity state changes. This is also here mainly because
+     * it is in the existing external Channel class.
+     * @param currentState The state to watch for transitions from. This should
+     *     always be populated by calling getConnectivityState immediately
+     *     before.
+     * @param deadline A deadline for waiting for a state change
+     * @param callback Called with no error when a state change, or with an
+     *     error if the deadline passes without a state change.
+     */
+    watchConnectivityState(currentState: ConnectivityState, deadline: Date|number, callback: (error?: Error) => void);
+    /**
+     * Create a call object. Call is an opaque type that is used by the Client
+     * class.
+     * @param method The full method string to request.
+     * @param deadline The call deadline
+     * @param host A host string override for making the request
+     * @param parentCall A server call to propagate some information from
+     * @param propagateFlags A bitwise combination of elements of grpc.propagate
+     *     that indicates what information to propagate from parentCall.
+     */
+    createCall(method: string, deadline: Date|number, host: string|null, parentCall: Call|null, propagateFlags: number|null): Call;
+}
+```
+
+### New `Client` Construction Options
+
+We will add the following options to the `Client` constructor's options map:
+
+ - `channelOverride`: a `Channel` instance. The `Client` will use this channel for communicating, and will ignore all other channel construction options.
```

Comment:
Nits: channelOverride => providedChannel ?

Reply:
I called it `channelOverride` because it overrides the `Client`'s default channel construction. But I'm not very attached to the name.